Sources/UI/Chat/Chat View Controller/ChatViewController.swift

```diff
@@ -402,8 +403,31 @@ extension ChatViewController {
             
         case let .itemsAdded(rows, reloadRow, forceToScroll, items):
             self.items = items
-            let needsToScroll = tableView.bottomContentOffset < bottomThreshold
-            tableView.stayOnScrollOnce = scrollEnabled && needsToScroll && !forceToScroll
+            
+            // A possible effective content height.
+            var effectiveContentHeight = tableView.frame.height // by default force to scroll.
+            
+            // Evaluate the possible effective content height for a signle message.
+            if rows.count == 1 {
+                var minMessageHeight = self.minMessageHeight
+                
+                if case .message(let message, []) = items[rows[0]] {
+                    if message.attachments.count > 1 {
+                        minMessageHeight = tableView.frame.height // always scroll for multiple attachments.
+                    } else if message.attachments.count == 1 {
+                        minMessageHeight += .attachmentPreviewMaxHeight
+                    } else if message.text.count > 70 {
```

Comment:
Will 70 characters always be 2 lines? In all screens?

Reply:
This is an approximate value for the smallest screens.